api/v1alpha1/gitrepository_types.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GitRepositorySpec struct {
+	Source GitRepositorySource `json:"source,omitempty"`
+	// GitURL is the base URL of Git server used for API calls.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.+$`
+	GitURL string `json:"gitURL"`
+	// SecretRef is the reference to secret that contain Git server credentials
+	// +kubebuilder:validation:Optional
+	SecretRef SecretReference `json:"secretRef"`
+}
+
+type GitRepositorySource struct {
+	// +kubebuilder:validation:Enum:=argocd;backstage;crossplane;gitea;nginx
```

Comment:
crossplane should be removed from the enum and we should keep what currently we support or point to a link = doc page enumerating them

Reply:
We will probably have to keep it since the reference implementation uses it. For now, I am keeping it because the purpose of this PR is to move the git functionality over from embedded git server to gitea.